app/controllers/api/storage.php

```diff
@@ -1159,12 +1181,21 @@
             );
         }
 
-        if (!empty($file->getAttribute('algorithm', ''))) {
-            if (empty($source)) {
-                $source = $deviceFiles->read($path);
-            }
-            $compressor = new GZIP();
-            $source = $compressor->decompress($source);
+        switch ($file->getAttribute('algorithm', 'none')) {
+            case 'zstd':
+                if (empty($source)) {
+                    $source = $deviceFiles->read($path);
+                }
+                $compressor = new Zstd();
+                $source = $compressor->decompress($source);
+                break;
+            case 'gzip':
+                if (empty($source)) {
+                    $source = $deviceFiles->read($path);
+                }
+                $compressor = new GZIP();
+                $source = $compressor->decompress($source);
+                break;
```

Comment:
Dont we need a default case?  
Because when we compress, we are defaulting to GZIP 
https://github.com/appwrite/appwrite/pull/3737/files#diff-02d3b2d220ab36c4733052fd1e3e017df738b47f7b3033dc8a93efbb7752436bR513